x-pack/plugins/aiops/public/hooks/use_search.ts

```diff
@@ -37,23 +43,43 @@ export const useSearch = (
     [dataView, uiSettings, savedSearch, filterManager]
   );
 
-  if (searchData === undefined || (aiopsListState && aiopsListState.searchString !== '')) {
-    if (aiopsListState?.filters && readOnly === false) {
-      const globalFilters = filterManager?.getGlobalFilters();
+  return useMemo(() => {
+    if (searchData === undefined || (aiopsListState && aiopsListState.searchString !== '')) {
+      if (aiopsListState?.filters && readOnly === false) {
+        const globalFilters = filterManager?.getGlobalFilters();
+
+        if (filterManager) filterManager.setFilters(aiopsListState.filters);
+        if (globalFilters) filterManager?.addFilters(globalFilters);
+      }
+
+      // In cases where the url state contains only a KQL query and not yet
+      // the transformed ES query we regenerate it. This may happen if we restore
+      // url state on page load coming from another page like ML's Single Metric Viewer.
+      let searchQuery = aiopsListState?.searchQuery;
```

Comment:
I am finding that a query passed from a job with a datafeed query is not getting applied e.g. view opens but this query is not applied:

<img width="1439" alt="image" src="https://github.com/elastic/kibana/assets/7405507/18e77fcf-4463-4c12-a143-ee653c98e534">

Compared to data in Lens with `agent.name: filebeat-vh2qr`:

<img width="1241" alt="image" src="https://github.com/elastic/kibana/assets/7405507/8014d039-de24-4c96-8eb2-814efa0f63ef">

Reply:
I was able to fix this in [fe922a9](https://github.com/elastic/kibana/pull/175289/commits/fe922a9724251e478137b043fd1942b4570dd8ad) and added functional tests covering it in [969b3ef](https://github.com/elastic/kibana/pull/175289/commits/969b3ef9d501cea4683aaaba5577fd5483c04945).